### PR TITLE
Bugfix: Various Future Sight / Doom Desire Fixes

### PR DESCRIPTION
**Pull request by aviettran**
Four separate issues. The commits are pretty tidy so you can view each individually for details.

### CODE REVIEW

File: data/scripts.ts [hunk 1]
```diff
@@ -335,7 +335,7 @@ export const Scripts: BattleScriptsData = {
 			[moveSteps[2], moveSteps[4]] = [moveSteps[4], moveSteps[2]];
 		}
 
-		this.setActiveMove(move, pokemon, targets[0]);
+		if (!notActive) this.setActiveMove(move, pokemon, targets[0]);
```
Comment (urkerab) [inline data/scripts.ts:338]: Maybe this really belongs in `useMoveInner` somewhere? (That way you wouldn't need it in `tryMoveHit` either.)

File: data/scripts.ts [hunk 2]
```diff
@@ -345,6 +345,7 @@ export const Scripts: BattleScriptsData = {
 				this.add('-fail', pokemon);
 				this.attrLastMove('[still]');
 			}
+			if (move.isFutureMove) return true;
 			return false;
```
Comment (urkerab) [inline data/scripts.ts:349]: We already have `this.NOT_FAIL` for other moves that don't double the power of Stomping Tantrum. Maybe this should say
```suggestion
			return hitResult === this.NOT_FAIL;
```
(with other related changes as necessary)

File: test/sim/moves/futuresight.js [hunk 3]
```diff
@@ -85,11 +85,12 @@ describe('Future Sight', function () {
 		battle.makeChoices();
 		battle.makeChoices();
 		battle.makeChoices();
+		battle.makeChoices();
```
Comment (urkerab) [inline test/sim/moves/futuresight.js:88]: I thought Future Sight hit at the end of the third turn, so you shouldn't need a fourth turn here. Are you sure you're really fixing the bug, and not just giving Wynaut a second chance to set Future Sight?

Comment (aviettran) [reply test/sim/moves/futuresight.js:88]: This is 3 turns. The third choice seems to be switching to Roggenrola after Shedinja faints.

File: data/scripts.ts (hunk 1 above)
Comment (aviettran) [reply data/scripts.ts:338]: The futuremove condition calls trySpreadMoveHit directly. Is it advisable to change this?

File: data/scripts.ts (hunk 2 above)
Comment (aviettran) [reply data/scripts.ts:349]: trySpreadMoveHit returns a boolean. We _could_ refactor it, but I think it's appropriate that it return true, i.e., futuresight _does_ succeed when it is used. Thoughts?

File: test/sim/moves/futuresight.js (hunk 3 above)
Comment (urkerab) [reply test/sim/moves/futuresight.js:88]: Ah, in that case I think a comment on the switch choice would make this clearer.

File: data/scripts.ts (hunk 2 above)
Comment (urkerab) [reply data/scripts.ts:349]: I'm not changing the return value of `trySpreadMoveHit` though am I?

File: data/scripts.ts (hunk 1 above)
Comment (urkerab) [reply data/scripts.ts:338]: (I'm insufficiently familiar with what effects can modify future moves to be able to contribute meaningfully to that question.)

File: test/sim/moves/futuresight.js (hunk 3 above)
Comment (aviettran) [reply test/sim/moves/futuresight.js:88]: resolved

File: data/scripts.ts (hunk 2 above)
Comment (aviettran) [reply data/scripts.ts:349]: Gotcha. Resolved.

File: test/sim/moves/futuresight.js (hunk 3 above)
Comment (urkerab) [reply test/sim/moves/futuresight.js]: Thanks!

File: data/scripts.ts (hunk 2 above)
Comment (urkerab) [reply data/scripts.ts]: Great, although personally I really don't like the `if (...) return true; return false;` especially when the condition actually has boolean type.

Comment (aviettran) [reply data/scripts.ts]: updated

**Review by Marty-D — APPROVED** [review]
👍

**Pull request merged**